gnomad/assessment/summary_stats.py

```diff
@@ -355,6 +356,185 @@ def get_tx_expression_expr(
+    lof_label_set: Optional[Set[str]] = None,
+    lof_flag_set: Optional[Set[str]] = None,
+    lof_no_flags: bool = False,
+    lof_any_flags: bool = False,
+    lof_loftee_combinations: bool = False,
+    additional_csq_sets: Optional[Dict[str, Set[str]]] = None,
+    additional_csqs: Optional[Set[str]] = None,
+    collapse_filters: bool = False,
+) -> Union[hl.expr.BooleanExpression, Dict[str, hl.expr.BooleanExpression]]:
+    """
+    Generate consequence filtering expression for summary stats.
+
+    :param t: Input Table/MatrixTable/StructExpression.
+    :param lof_csq_set: Set of LoF consequence strings. Default is None.
+    :param lof_label_set: Set of LoF consequence labels. Default is None.
+    :param lof_flag_set: Set of LoF consequence flags. Default is None.
+    :param lof_no_flags: Whether to filter to LoF variants with no flags. Default is
+        False.
+    :param lof_any_flags: Whether to filter to LoF variants with any flags. Default is
+        False.
+    :param lof_loftee_combinations: Whether to add combinations of LOFTEE and
+        consequence type filters. Default is False.
+    :param additional_csq_sets: Dictionary containing additional consequence sets.
+        Default is None.
+    :param additional_csqs: Set of additional consequences to keep. Default is None.
+    :param collapse_filters: Whether to collapse all filters into a single expression.
+        Default is False.
+    :return: BooleanExpression or Dict of BooleanExpressions for filtering consequences.
+    """
+    # Set up filters for specific consequences or sets of consequences.
+    csq_filters = {
+        **({"csq_set_lof": lof_csq_set or {}}),
+        **({f"csq_set_{l}": c for l, c in (additional_csq_sets or {}).items()}),
+        **({f"lof_csq_{c}": {c} for c in lof_csq_set or []}),
+        **({f"csq_{c}": {c} for c in additional_csqs or []}),
+    }
+
+    def _create_filter_by_csq(
+        t: Union[hl.Table, hl.MatrixTable],
+        csq_set: Union[Set, List, hl.expr.CollectionExpression],
+    ) -> hl.expr.BooleanExpression:
+        """
+        Create filtering expression for a set of consequences.
+
+        :param t: Input Table/MatrixTable.
+        :param csq_set: Set of consequences to filter.
+        :return: BooleanExpression for filtering by consequence.
+        """
+        if not isinstance(csq_set, hl.expr.CollectionExpression):
+            csq_set = hl.set(csq_set)
+
+        return csq_set.contains(t.most_severe_csq)
+
+    # Create filtering expressions for each consequence set.
+    ss_filter_expr = {
+        filter_name: _create_filter_by_csq(t, csq_set)
+        for filter_name, csq_set in csq_filters.items()
+    }
+
+    # Add filtering expressions for LoF consequence labels.
+    lof_labels = {
+        f"lof_{lof_label}": hl.or_else(t.lof == lof_label, False)
+        for lof_label in lof_label_set or []
+    }
+
+    # Add filtering expressions for LoF consequence flags.
+    lof_flags = {
```

Comment:
Even thought we're using it this time, do you think it would be nice to have `LOFTEE_FLAGS` in vep.py?

```commandline
LOFTEE_FLAGS = {
    'NAGNAG_SITE',
    'NO_EXON_NUMBER',
    'NON_CAN_SPLICE',
    'PHYLOCSF_UNLIKELY_ORF',
    'PHYLOCSF_WEAK',
    'SINGLE_EXON',
}
"""
Constant that contains LOFTEE flags.
"""
```

Reply:
Removed that option for now, so we can just add if/when needed